CMakeLists.txt

```diff
@@ -42,18 +42,24 @@ else()
                   -fsanitize=alignment -fsanitize=null -fsanitize=address)
     add_compile_options(${SAN_FLAGS})
     link_libraries(${SAN_FLAGS})
+    # A non-zero optimization level is desired in debug mode, but allow overriding it nonetheless
+    # TODO: this overrides anything previously set... that's a bit sloppy!
+    set(CMAKE_C_FLAGS_DEBUG "-g -Og -fno-omit-frame-pointer -fno-optimize-sibling-calls" CACHE STRING "" FORCE)
   endif()
 
   if(MORE_WARNINGS)
-    add_compile_options(-Werror -Wextra -Wno-type-limits
-                        -Wno-sign-compare -Wvla -Wformat -Wformat-security -Wformat-overflow=2
-                        -Wformat-truncation=1 -Wformat-y2k -Wswitch-enum -Wunused
-                        -Wuninitialized -Wunknown-pragmas -Wstrict-overflow=5
-                        -Wstringop-overflow=4 -Walloc-zero -Wduplicated-cond
-                        -Wfloat-equal -Wshadow -Wcast-qual -Wcast-align -Wlogical-op
-                        -Wnested-externs -Wno-aggressive-loop-optimizations -Winline
-                        -Wundef -Wstrict-prototypes -Wold-style-definition
-                        -Wno-unknown-warning-option -Wno-tautological-constant-out-of-range-compare)
+    add_compile_options(-Werror -Wextra
+                        -Walloc-zero -Wcast-align -Wcast-qual -Wduplicated-branches -Wduplicated-cond
+                        -Wfloat-equal -Winline -Wlogical-op -Wnested-externs -Wnull-dereference
+                        -Wold-style-definition -Wshift-overflow=2 -Wstrict-overflow=5
+                        -Wstrict-prototypes -Wstringop-overflow=4 -Wundef -Wuninitialized -Wunused
+                        -Wshadow # TODO: -Wshadow=compatible-local ?
```

Comment:
I don't see any warnings caused by `-Wshadow`; is it a problem?

Reply:
No, but I've hit it several times during development, and have had to come up with somewhat contrived variable names then.

Reply:
That doesn't necessarily sound like a problem (if I have an outer `const char *s` I probably still don't want an inner `struct Symbol s`), but if you think the names are too contrived, sure, `-Wshadow=compatible-local` is fine.

Reply:
`s` is a bad variable name anyway, but an example of when it can be problematic is a global `symbols` that gets shadowed by a local `symbols` when attempting to resolve a reference.

I don't have any strong feelings about it, so we should leave it a TODO for now, and come back to it the next time it becomes relevant.

Reply:
I think that's a good example of why global variables should have longer more precise names; I have multiple guesses for what a global `symbols` could be.

Reply:
What should it be, then? That's the storage for all the symbols.